.github/workflows/scripts/util.js

```diff
@@ -0,0 +1,150 @@
+// @ts-check
+
+module.exports = {
+  addLabelIfNotExists,
+  getLabels,
+  group,
+  hasLabel,
+  removeLabelIfExists,
+};
+
+/** @type {Map<number, Set<string>>} */
+const labelCache = new Map();
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments['github']} github
+ * @param {import('github-script').AsyncFunctionArguments['context']} context
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @param {string} name
+ * @returns {Promise<void>}
+ */
+async function addLabelIfNotExists(github, context, core, name) {
+  await group(`addLabelIfNotExists("${name}")`, async () => {
+    if (!context.payload.pull_request) {
+      throw new Error("May only run in context of a pull request");
+    }
+
+    if (await hasLabel(github, context, name)) {
+      console.log(`Already has label '${name}'`);
+      return;
+    }
+
+    core.notice(`Adding label '${name}'`);
+
+    const prNumber = context.payload.pull_request.number;
+    const { data: labels } = await github.rest.issues.addLabels({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      issue_number: prNumber,
+      labels: [name],
+    });
+
+    const labelNames = new Set(labels.map((l) => l.name));
+    labelCache.set(prNumber, labelNames);
+ });
+}
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments['github']} github
+ * @param {import('github-script').AsyncFunctionArguments['context']} context
+ * @returns {Promise<Set<string>>} Set of labels associated with the current PR
+ */
+async function getLabels(github, context) {
+  return await group(`getLabels()`, async () => {
+    if (!context.payload.pull_request) {
+      throw new Error("May only run in context of a pull request");
+    }
+
+    const prNumber = context.payload.pull_request.number;
+
+    let labelNames = labelCache.get(prNumber);
+    if (!labelNames) {
+      const { data: labels } = await github.rest.issues.listLabelsOnIssue({
```

Comment:
Can reason we cannot pull this from the context object? `context.payload.pull_request.labels`

Reply:
I will need to see if this is available inside `github-script`.  Intellisense isn't showing it, but I believe the intellisense may be incomplete.

Reply:
This is probably available in the context, but we'll need to make sure it's populated even in fork PRs.